test-support/page-object/properties/click-on-text.js

```diff
@@ -45,11 +51,22 @@ export function clickOnText(selector, options = {}) {
     isDescriptor: true,
 
     value(textToClick) {
-      wait().then(() => {
-        var actualSelector = findChildElement(this, selector, textToClick, options) || findElement(this, selector, textToClick, options);
+      var context = getContext(this);
+
+      if (context) {
+        Ember.run(() => {
+          var fullSelector = actualSelector(this, selector, textToClick, options);
+
+          context.$(fullSelector).click();
+        });
+      } else {
```

Comment:
Yes, sorry about that. Codacy generates these comments automatically based on some predefined rules and it seems it has some issues.

I'll turn off automatic comments from Codacy, it's not that useful.